src/Algebra/Properties/CancellativeCommutativeSemiring.agda

```diff
@@ -0,0 +1,51 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of operations in CancellativeCommutativeSemiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (CancellativeCommutativeSemiring)
+open import Algebra.Definitions using (AlmostRightCancellative)
+open import Data.Product using (_,_)
+open import Data.Sum.Base using (_⊎_; inj₁; inj₂)
+open import Function using (_$_)
+open import Relation.Binary using (Decidable)
+import Relation.Binary.Reasoning.Setoid as EqReasoning
+open import Relation.Nullary using (yes; no)
+open import Relation.Nullary.Negation using (contradiction)
+
+module Algebra.Properties.CancellativeCommutativeSemiring
+  {a ℓ} (R : CancellativeCommutativeSemiring a ℓ)
+  (open CancellativeCommutativeSemiring R) where
+
+open EqReasoning setoid
+open import Algebra.Consequences.Setoid setoid
+  using (comm+cancelˡ-nonZero⇒cancelʳ-nonZero)
+open import Algebra.Properties.CommutativeSemigroup *-commutativeSemigroup
+  using (x∙yz≈y∙xz)
+open import Algebra.Properties.Semiring.Divisibility semiring
+
+*-cancelʳ-nonZero : AlmostRightCancellative _≈_ 0# _*_
+*-cancelʳ-nonZero = comm+cancelˡ-nonZero⇒cancelʳ-nonZero *-comm 0# *-cancelˡ-nonZero
+
+x*y≈0⇒x≈0⊎y≈0 : Decidable _≈_ → ∀ {x y} → x * y ≈ 0# → x ≈ 0# ⊎ y ≈ 0#
```

Comment:
By the naming changes you've made, this should be called `xy≈0⇒x≈0⊎y≈0`

Reply:
Aren't we using `_∨_` in names?

Reply:
Renamied to `xy≈0⇒x≈0∨y≈0`.